core/services/ocr2/plugins/ocr2vrf/fixed_blockhash_provider.go

```diff
@@ -0,0 +1,98 @@
+package ocr2vrf
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/pkg/errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/smartcontractkit/chainlink/core/chains/evm/client"
+	evmtypes "github.com/smartcontractkit/chainlink/core/chains/evm/types"
+	"github.com/smartcontractkit/ocr2vrf/types"
+)
+
+// fixedBlockhashProvider returns blockhashes with fixed-sized windows relative to current block height
+type fixedBlockhashProvider struct {
+	client client.Client
+	// start block = current head - lookbackBlocks
+	lookbackBlocks uint
+	// number of blocks to query in a batch
+	batchSize uint
+}
+
+var _ types.Blockhashes = (*fixedBlockhashProvider)(nil)
+
+func NewFixedBlockhashProvider(client client.Client, lookbackBlocks, batchSize uint) types.Blockhashes {
+	return &fixedBlockhashProvider{
+		client,
+		lookbackBlocks,
+		batchSize,
+	}
+}
+
+func (b *fixedBlockhashProvider) OnchainVerifiableBlocks(
+	ctx context.Context,
+) (startHeight uint64, hashes []common.Hash, err error) {
+	toBlock, err := b.CurrentHeight(ctx)
+	if err != nil {
+		return 0, nil, errors.Wrap(err, "current height")
+	}
+	fromBlock := toBlock - uint64(b.lookbackBlocks)
+	var reqs []rpc.BatchElem
+
+	for i := fromBlock; i <= toBlock; i++ {
+		req := rpc.BatchElem{
+			Method: "eth_getBlockByNumber",
+			Args:   []interface{}{hexutil.EncodeBig(big.NewInt(int64(i))), true},
+			Result: &evmtypes.Head{},
```

Comment:
I'd like to see what errors you're getting for using `&types.Header{}` instead, since it's more correct. `evmtypes.Head` _seems_ to work because it has some of the fields we need, but is technically the wrong type to use here.